### PR TITLE
Adding TileConfiguration parser and other improvements

### PR DESCRIPTION
**Pull request by dzenanz**

### CODE REVIEW

**Comment by dzenanz** [comment]
The tests pass on Windows (except Ti7 test, which has been failing before this PR). OMC and RGB tests segfault on Linux. I will investigate that sometime later.

File: src/itkParseTileConfiguration.cxx [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+      }
+    }
+  return temp;
+}
+
+static double_conversion::StringToDoubleConverter stringConverter(
+  double_conversion::StringToDoubleConverter::ALLOW_TRAILING_JUNK |
+  double_conversion::StringToDoubleConverter::ALLOW_LEADING_SPACES |
+  double_conversion::StringToDoubleConverter::ALLOW_TRAILING_SPACES,
+  0.0, std::numeric_limits<double>::quiet_NaN(), nullptr, nullptr);
```
Comment (thewtex) [inline src/itkParseTileConfiguration.cxx]: :+1:

File: src/itkParseTileConfiguration.cxx [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+
+namespace itk
+{
+std::vector< std::vector< Tile< 2 > > >
+ParseTileConfiguration2D( const std::string pathToFile )
+{
+  constexpr unsigned Dimension = 2;
+  using PointType = itk::Point< double, Dimension >;
+  using TileRow = std::vector< Tile< Dimension > >;
```
Comment (thewtex) [inline src/itkParseTileConfiguration.cxx]: Please use these named types in the class and in the example so their meaning is known. `std::vector< std::vector< itk::Tile< 2 > >` was opaque.

File: src/itkParseTileConfiguration.cxx [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+  using TileLayout = std::vector< TileRow >;
```
Comment (thewtex) [inline src/itkParseTileConfiguration.cxx]: Maybe `TileLayout2D` or other to make the 2D content more explicit.

**Comment by thewtex** [comment]
> The tests pass on Windows (except Ti7 test, which has been failing before this PR). OMC and RGB tests segfault on Linux. I will investigate that sometime later.

Please create issues in the issue tracker for known failing tests, etc.

**Comment by dzenanz** [comment]
Added #76 to track Ti7 test.

**Comment by dzenanz** [comment]
The segfaults on Linux occur due to Windows-style line endings in TileConfiguration.txt files 😄

I will fix that after lunch.

**Comment by thewtex** [comment]
Thanks, @dzenanz !

**Pull request merged**